src/interface/src/styleguide/filter-dropdown/filter-dropdown.component.ts

```diff
@@ -0,0 +1,127 @@
+import {
+  Component,
+  EventEmitter,
+  Input,
+  OnInit,
+  Output,
+  ViewChild,
+} from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { MatIconModule } from '@angular/material/icon';
+import { MatMenuModule } from '@angular/material/menu';
+import { MatCheckboxModule } from '@angular/material/checkbox';
+import { InputFieldComponent } from '@styleguide';
+import { FormsModule } from '@angular/forms';
+import { ButtonComponent } from '../button/button.component';
+import { SearchBarComponent } from '../search-bar/search-bar.component';
+import { Subject } from 'rxjs';
+/**
+ * Filter dropdown that lets user select one or multiple strings as part of a search
+ */
+@Component({
+  selector: 'sg-filter-dropdown',
+  standalone: true,
+  imports: [
+    CommonModule,
+    MatIconModule,
+    MatMenuModule,
+    MatCheckboxModule,
+    InputFieldComponent,
+    SearchBarComponent,
+    FormsModule,
+    ButtonComponent,
+  ],
+  templateUrl: './filter-dropdown.component.html',
+  styleUrls: ['./filter-dropdown.component.scss'],
+})
+export class FilterDropdownComponent implements OnInit {
+  @ViewChild(SearchBarComponent) searchbar!: SearchBarComponent;
+
+  /**
+   * Accepts the name of a Material Icon to be displayed at the left of the menu trigger
+   */
+  @Input() leadingIcon = '';
+  /**
+   * Determines whether a search bar is present for filtering menu options
+   */
+  @Input() hasSearch = true;
+  @Input() disabled = false;
+  /**
+   * The label text displayed for the menu
+   */
+  @Input() menuLabel!: string;
+  /**
+   * Items displayed in the menu
+   */
+  @Input() menuItems!: string[];
```

Comment:
Im wondering if an array of strings is good enough for this, or we really need an array of key,values